tools/buildsys/src/gomod.rs

```diff
@@ -0,0 +1,176 @@
+                export GOSUMDB={gosumdb} &&
+                export GOPRIVATE={goprivate} &&
+                go list -mod=readonly ./... >/dev/null && go mod vendor &&
+                popd &&
+                tar czf {output} {moddir} && 
+                rm -rf {moddir}",
+                input = path_arg.to_string_lossy(),
+                moddir = mod_dir.to_string_lossy(),
+                goproxy = goproxy,
+                gosumdb = gosumdb,
+                goprivate = goprivate,
+                output = output_path_arg.to_string_lossy(),
+            ),
+        };
+        docker_go(root_dir, &args)?;
+
+        Ok(())
+    }
+}
+
+fn extract_file_name(url: &str) -> Result<PathBuf> {
+    let parsed = reqwest::Url::parse(url).context(error::InputUrlSnafu { url })?;
+    let name = parsed
+        .path_segments()
+        .context(error::InputFileBadSnafu { path: url })?
+        .last()
+        .context(error::InputFileBadSnafu { path: url })?;
+    Ok(name.into())
+}
+
+struct DockerGoArgs<'a> {
+    module_path: &'a Path,
+    sdk_image: String,
+    go_mod_cache: &'a Path,
+    command: String,
+}
+
+/// Run `docker-go` with the specified arguments.
+fn docker_go(root_dir: &Path, dg_args: &DockerGoArgs) -> Result<Output> {
+    let args = vec![
+        "--module-path",
+        dg_args
+            .module_path
+            .to_str()
+            .context(error::InputFileSnafu)?,
+        "--sdk-image",
+        &dg_args.sdk_image,
+        "--go-mod-cache",
+        dg_args
+            .go_mod_cache
+            .to_str()
+            .context(error::InputFileSnafu)?,
+        "--command",
+        &dg_args.command,
+    ];
+    let arg_string = args.join(" ");
+    let program = root_dir.join("tools/docker-go");
+    println!("program: {}", program.to_string_lossy());
+    let output = cmd(program, args)
+        .stderr_to_stdout()
+        .stdout_capture()
+        .unchecked()
+        .run()
+        .context(error::CommandStartSnafu)?;
+
+    let stdout = String::from_utf8_lossy(&output.stdout);
+    println!("{}", &stdout);
+    ensure!(
+        output.status.success(),
+        error::DockerExecutionSnafu { args: arg_string }
+    );
+    Ok(output)
+}
+
+/// Run `go env` with the specified argument.
+fn go_env(var: &str) -> Option<String> {
+    let args = vec!["env", var];
+    let output = match cmd("go", args)
+        .stderr_to_stdout()
+        .stdout_capture()
+        .unchecked()
+        .run()
+    {
+        Ok(v) => v,
+        Err(_) => return None,
+    };
+    let stdout = String::from_utf8_lossy(&output.stdout);
+    println!("{}", &stdout);
+    output.status.success().then(|| stdout.to_string())
+}
+
+/// Retrieve a BUILDSYS_* variable that we expect to be set in the environment,
+/// and ensure that we track it for changes, since it will directly affect the
+/// output.
+fn getenv(var: &str) -> Result<String> {
+    println!("cargo:rerun-if-env-changed={}", var);
```

Comment:
I think we are not supposed to do this anymore and instead use `bottlerocket-variant`, and probably this function? https://github.com/bottlerocket-os/bottlerocket/blob/037d0e2798a7e8d4a12d0fb35d3c266db3d74f25/sources/bottlerocket-variant/src/lib.rs#L178

I'll let others comment on this since I'm not an expert about when the library linked should be used.

Reply:
Good question: This was a chunk of code from Sam's initial PR and I assumed would force a rerun if any of the go mod control env variables changed. If there's a better way to do this, let me know!